build.go

```diff
@@ -320,9 +322,13 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 
 	if !opts.TrustBuilder {
 		if lifecycleImageSupported(imgOS, lifecycleVersion) {
+			lifecycleImageRepo := opts.LifecycleImageRepo
+			if lifecycleImageRepo == "" {
+				lifecycleImageRepo = internalConfig.DefaultLifecycleImageRepo
+			}
 			lifecycleImage, err := c.imageFetcher.Fetch(
 				ctx,
-				fmt.Sprintf("%s:%s", LifecycleImageRepo, lifecycleVersion.String()),
+				fmt.Sprintf("%s:%s", lifecycleImageRepo, lifecycleVersion.String()),
```

Comment:
I'm curious how this feature of setting a specific lifecycle repo plays with specific images and lifecycle versions.

For example, say I have a very specific SHA I wanted to test `myorg/mylifecycle@sha256:...`  or say I wanted to run an image that is tagged differently `myorg/mylifecycle:tomato` (because I like naming stuff :)).

Given the current simple appending strategy it would appear that the above cases would not be supported. I wonder if we can easily enable them by [parsing the input](https://pkg.go.dev/github.com/google/go-containerregistry/pkg/name#ParseReference) to determine if a [`Digest`](https://pkg.go.dev/github.com/google/go-containerregistry/pkg/name#Digest) or [`Tag`](https://pkg.go.dev/github.com/google/go-containerregistry/pkg/name#Tag) was provided.

In my mind it would look something like:

| input | builder lifecycle version | image used
|---      |---                                     |---
| myorg/lifecycle | 0.10.2 | myorg/lifecycle:0.10.2
| myorg/lifecycle@sha256:123... | 0.10.2 | myorg/lifecycle@sha256:123...
| myorg/lifecycle:tomato | 0.10.2 | myorg/lifecycle:tomato

Reply:
The reason I didn't enable this was because it might cause conflicts with the builder's lifecycle version. Would we be saying that we will be overriding the lifecycle version set by the builder.toml in favor of this lifecycle image? And how would we handle compat issues. As such, I thought that at first supporting just a different repo/registry (which may be an internal mirror) would be the easiest. I think there were some suggestions about moving away from the lifecycle version to the supported platform version instead in which case I think we can relax the constraints and provide some better handling here.

This may be out of the scope of this PR/issue given that the original issue was just to support mirrors, but if it is just an easy hardcoded override I think we should be able to support the above usercase.

Reply:
I don't think the lifecycle version disparity between phases is a huge concern given that it's a contract (Platform API) that is being adhered to. For example, this [hardcoded version in tekton](https://github.com/tektoncd/catalog/blob/master/task/buildpacks-phases/0.1/buildpacks-phases.yaml#L110) could be different than that provided by the builder.

@ekcasey / @natalieparellano correct me if I'm wrong.

Furthermore, I would suspect that the use of this option is an "advanced" option meaning that the user would have more of an idea on what ramifications may be by forcing a specific lifecycle image.

Reply:
Gotcha - so the expected changes is as you described in the table above -

If the user gives just the image repo - we use the lifecycle version as the tag. If they specify a image repo + tag we use that. Same for image repo + SHAs?

Reply:
Just my two cents but I think we should interpret `--lifecycle-image myorg/lifecycle` as `--lifecycle-image myorg/lifecycle:latest`. Folks are very used to the implicit `:latest` and doing anything else might be surprising.

Reply:
Hmm - makes sense. Should I rename the flag from image repo to image/image tag then?

Reply:
`--lifecycle-image` makes the most sense to me

Reply:
> Just my two cents but I think we should interpret `--lifecycle-image myorg/lifecycle` as `--lifecycle-image myorg/lifecycle:latest`. Folks are very used to the implicit `:latest` and doing anything else might be surprising. 

I'd be in favor of that due to the positive UX. I was more concerned with appending lifecycle versions tags. If that's no longer the case then my suggestions above are moot.

Reply:
In general probably we'd want to recommend using the same lifecycle version as the provided builder (if possible) but as long as the lifecycle supports the same api versions as the builder, it should work.